### ISSUE TITLE
Django: InfractionSerializer.validate/2 should live in model constraints instead

### ISSUE THREAD

**Issue by jchristgit** [python-discord/site#217]

**Comment by MarkKoz** [comment]
Did you already figure out how to do this?

I wanted to do it this way from the start, but I couldn't figure this out. IIRC, DRF does handle field-level validators in the model, but it does not handle object-level validation in the model (i.e. it doesn't call any of the `clean()` methods). `clean()` could be called inside the serialiser's `validate()`, but I believe that would mean we have to handle the `ValidationError`s ourselves because they're not the same as `serializers.ValidationError`.

**Comment by jchristgit** [comment]
Hmm that sounds odd. I haven't given this a try yet but I'll try to check it out tomorrow.

**Comment by MarkKoz** [comment]
On second thought I was thinking of validation while you were probably thinking of the new constraints in 2.2. Maybe it doesn't apply, not familiar with constraints in Django.

**Comment by jchristgit** [comment]
Yes that is the case. Constraints, database constraints, cannot be ignored.

**Comment by jchristgit** [comment]
The database knows it all.
